File: app/src/main/java/com/example/floating/bakingapp/di/AppModule.java
package com.example.floating.bakingapp.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by apple on 14/01/2018.
 */

@Module
public class AppModule {

    private Application application;

    public AppModule(Application application){
        this.application = application;
    }

    @Provides
    @Singleton
    Application providesApplication(){
        return application;
    }

}

File: app/src/main/java/com/example/floating/bakingapp/di/AppModule.kt
package com.example.floating.bakingapp.di

import android.app.Application

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Created by apple on 14/01/2018.
 */

@Module
class AppModule(private val application: Application) {

    @Provides
    @Singleton
    internal fun providesApplication(): Application {
        return application
    }

}
